rust/src/libnm_dbus/connection/ieee8021x.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::HashMap;
+use std::convert::TryFrom;
+
+use serde::Deserialize;
+
+use crate::{connection::DbusDictionary, ErrorKind, NmError};
+
+const GLIB_FILE_PATH_PREFIX: &str = "file://";
+
+#[derive(Debug, Clone, PartialEq, Default, Deserialize)]
+#[serde(try_from = "DbusDictionary")]
+pub struct NmSetting8021X {
```

Comment:
Please add the `non-exhaustive` as we do not want to include new structs without it.